assets/app/scripts/controllers/pod.js

```diff
@@ -47,6 +47,35 @@ angular.module('openshiftConsole')
       $scope.logCanRun = !(_.includes(['New', 'Pending', 'Unknown'], pod.status.phase));
     };
 
+    var setContainerVars = function() {
+      if(!$scope.pod) {
+        return;
+      }
+      var containerStatus = _.find($scope.pod.status.containerStatuses, { name: $scope.logOptions.container });
+      var state = _.get(containerStatus, 'state');
+      var statusKey = _.head(_.keys(state));
+      var knownKey = _.includes(['running', 'waiting', 'terminated'], statusKey) ? statusKey : '';
+      var lastState = _.get(containerStatus, 'lastState');
+      var lastStatusKey = _.head(_.keys(lastState));
+      var isWaiting = _.isEqual(_.head(_.keys(_.get(containerStatus, 'state'))), 'waiting');
+
+      angular.extend($scope, {
+        containerStatusKey: knownKey,
+        containerExitReason: _.get(state, [statusKey, 'reason'])
```

Comment:
containerStateReason? It's not always an exit reason. You can have reasons for waiting containers.